fs-cache/src/memory_limited_storage.rs

```diff
@@ -0,0 +1,273 @@
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::time::SystemTime;
+
+use data_error::{ArklibError, Result};
+use linked_hash_map::LinkedHashMap;
+
+pub struct MemoryLimitedStorage<K, V> {
+    /// Label for logging
+    label: String,
+    /// Path to the underlying folder where data is persisted
+    path: PathBuf,
+    /// In-memory LRU cache combining map and queue functionality
+    memory_cache: LinkedHashMap<K, V>,
```

Comment:
Nit: I think this comment is slightly incorrect. `LinkedHashMap` is not the cache itself.